www/settime

```diff
@@ -0,0 +1,132 @@
+<?php
+
+include_once "session-start.php";
+include_once "util.php";
+include_once "pagetpl.php";
+include_once "login-check.php";
+
+function sendResponse($statmsg, $errmsg, $detail)
+{
+    global $xml;
+
+    if ($xml)
+    {
+        header("Content-Type: text/xml");
+        header("Expires: Mon, 26 Jul 1997 05:00:00 GMT");
+        header("Cache-Control: no-store, no-cache, must-revalidate");
+
+        echo "<?xml version=\"1.0\" encoding=\"UTF-8\" ?><response>"
+            . ($statmsg ? "<label>$statmsg</label>" : "")
+            . ($errmsg ? "<error>$errmsg</error>" : "")
+            . "</response>";
+    }
+    else
+    {
+        smallPageHeader("Update Your Time Vote");   // Do we need this?
+        if ($errmsg)
+            echo "<span class=errmsg>$errmsg</span><p>";
+
+        echo $detail;
+        smallPageFooter();
+    }
+
+    exit();
+}
+
+// connect to the database
+include_once "dbconnect.php";
+$db = dbConnect();
+if ($db == false)
+    sendResponse("Not Saved", "An error occurred connecting to the database. "
+                 . "Please try again later.", false);
+
+// get the request parameters
+$game = get_req_data('game');
+$newtime = get_req_data('newtime');
+$xml = isset($_REQUEST['xml']);
+
+// make sure there's a game
+if ($game == "")
+    sendResponse("Not Saved", "No game was specified.", false);
+
+// make sure we process any persistent login state
+$userid = checkPersistentLogin();
+if (!$userid && !$xml && !logged_in(true))
+    exit();
+
+// make sure we're logged in
+if (!$userid)
+    sendResponse("Not Saved", "To submit a time, please log in.", false);
+
+// make sure the game is valid
+$qgame = mysql_real_escape_string($game, $db);
+$result = mysql_query("select id from games where id = '$qgame'", $db);
+if (mysql_num_rows($result) == 0)
+    sendResponse("Not Saved", "The specified game was not found.", false);
+
+// if there's an existing time for this user for this game, simply update
+// the time; otherwise add a new row
+
+$result = mysql_query("lock tables playertimes write", $db);
```

Comment:
why the lock?

Reply:
That's copied from the setrating page. I don't know much about it.

Reply:
I don't know why it's there either :D @dfabulich ?

Reply:
That lock has been in `setrating` since we inherited the code from MJR. Reading the comments, it looks like he was using the lock to do an "upsert", which nowadays we would do with `INSERT … ON DUPLICATE KEY UPDATE`.